### PR TITLE
Add warnings if observation time differs from image time

### PR DESCRIPTION
**Pull request by dgarciabriseno**
# Summary

This update implements a warning message to show you if the image in the viewport >= 6 hours from your observation time. Prior tot his patch, helioviewer had a more subtle method which is to color the date in the left sidebar from green to red over a time delta of 12 hours. That is:
- if the image and observation time are the same, then the text is green.
- if the image and observation time are >= 12 hours apart, the text is red

With this update, at the 6 hour mark, we will show a warning that the image is beginning to get farther away from the observation time.

# Testing

Enter the browsers and the output modes (minimal, normal, embed) this change was tested.

| *browser* | *normal* | *minimal* | *embed* | 
| ------- | -------- | -------- | -------- |
| firefox | tested | tested | tested |
| chrome  | tested | tested | tested |
| safari  | tested | tested | tested |
| edge    | tested | tested | tested |

### CODE REVIEW

File: resources/js/Image/JP2Image.js [hunk 1]
```diff
@@ -92,6 +95,92 @@ var JP2Image = Class.extend(
...
+     * @param {getClosestImageResponse} metadata
+     */
+    _notifyIfStaleImage: function (metadata) {
+        // Get the image's timestamp and the current observation time
+        let imageDate = Date.parseUTCDate(metadata.date);
+        let obstTime = helioviewer.timeControls.getDate();
+        // Get the time difference between the two times in seconds
+        let delta = Math.abs(imageDate.getTime() - obstTime.getTime()) / 1000;
+        // Get the preset threshold in seconds
+        let threshold = Helioviewer.userSettings.get("thresholds.obstime");
```
Comment (mudhoney) [inline resources/js/Image/JP2Image.js]: This seems like a thing for Config, not for user settings,

File: resources/js/Image/JP2Image.js [hunk 2]
```diff
@@ -92,6 +95,92 @@ var JP2Image = Class.extend(
...
+        // notifications for the same layer.
+        let notification = this._notification ? await this._notification : null;
+
+        // This promise resolves to the jgrowl message in the DOM that is
+        // associated with this notification.
+        this._notification = new Promise(async (resolve) => {
+            // Create the notification message
+            let message = "The " + name + " layer is " + humanReadableNumSeconds(delta) + " away from your observation time";
+            // Create the css class that will be assigned to this notification
+            let group = name.replace(" ", "-");
```
Comment (mudhoney) [inline resources/js/Image/JP2Image.js:154]: You might not need group anymore

File: resources/js/UI/ImagePresets.js [hunk 3]
```diff
@@ -206,7 +205,7 @@ var UserLayersPresets = Class.extend({
 
 	},
 
-	_loadData: function(el){
+	_loadData: async function(el){
         var date = $(el).data('date');
```
Comment (mudhoney) [inline resources/js/UI/ImagePresets.js:209]: indentation

File: resources/js/Image/JP2Image.js (hunk 1 above)
Comment (dgarciabriseno) [reply resources/js/Image/JP2Image.js]: You're right, moving it to config

File: resources/js/Image/JP2Image.js (hunk 2 above)
Comment (dgarciabriseno) [reply resources/js/Image/JP2Image.js:154]: Never mind, I do need group to remove duplicates.

File: resources/js/UI/ImagePresets.js (hunk 3 above)
Comment (dgarciabriseno) [reply resources/js/UI/ImagePresets.js:209]: Fixed

File: resources/js/Image/JP2Image.js (hunk 1 above)
Comment (dgarciabriseno) [reply resources/js/Image/JP2Image.js]: Moved to serverSettings

**Review by mudhoney — APPROVED** [review]

**Pull request merged**